pkg/controller/job/job_controller_test.go

```diff
@@ -1840,8 +1840,6 @@ func TestSyncPastDeadlineJobFinished(t *testing.T) {
 	sharedInformerFactory.Start(ctx.Done())
 	sharedInformerFactory.WaitForCacheSync(ctx.Done())
 
-	go manager.Run(ctx, 1)
```

Comment:
Is the change about replacing `go manager.Run(ctx, 1)` with invocations of `syncJob` necessary? IIUC the issue is that the job has start time shifted by one second to the past and this is exactly the value for `ActiveDeadlineSeconds`, so avoiding being on the edge should be enough. If this is the case then I suggest to make the fix minimal and only keep setting `ActiveDeadlineSeconds` to 2 and doing the sleep `manager.clock.Sleep(2 * time.Second)`.

Reply:
>Is the change about replacing go manager.Run(ctx, 1) with invocations of syncJob necessary?

@mimowo not really a mandatory change. But it enables a bit more control on performing the sync is what my thought behind the change was. Do you think that is an overkill ?

Reply:
I wouldn't call it overkill as this is a simple change, but it makes harder to reason about what is needed to fix the test. 

That being said, the change is reasonable to me in a unit test, so I would suggest one of the two approaches:
1. Leave as is, but make it clear in the issue description what is the needed change and what is a drive-by change to make debugging / reasoning about the test simpler. Preferably split the changes into two commits withing the PR.
2. Make only necessary changes in this PR, but open a follow up PR to improve debuggability / clarity of the test.

If you make changes to improve debuggability of the test you may want as well to return the `err` here https://github.com/kubernetes/kubernetes/blob/7eb9a75bf914428e7579d2497722f71be3ef1812/pkg/controller/job/job_controller_test.go#L1900 and log it there: https://github.com/kubernetes/kubernetes/blob/7eb9a75bf914428e7579d2497722f71be3ef1812/pkg/controller/job/job_controller_test.go#L1908

Reply:
/Ack.

Reply:
@mimowo  I have taken up your suggestion and did what you suggested in option 1. PTAL when possible.

Reply:
Please revert this.
If you call syncJob directly, then we are not testing the reinsertions into the workqueue, which is where the original bug was initially.